diff --git a/hwy/detect_compiler_arch.h b/hwy/detect_compiler_arch.h
index 5d08c61c..38f4cd88 100644
--- a/hwy/detect_compiler_arch.h
+++ b/hwy/detect_compiler_arch.h
@@ -226,7 +226,7 @@
 #define HWY_ARCH_X86_32 0
 #endif
 
-#if defined(__x86_64__) || defined(_M_X64)
+#if (defined(__x86_64__) || defined(_M_X64)) && !defined(__arm64ec__) && !defined(_M_ARM64EC)
 #define HWY_ARCH_X86_64 1
 #else
 #define HWY_ARCH_X86_64 0
